test/fixtures/plugin-grpc.js

```diff
@@ -0,0 +1,414 @@
+ * as well as serialize and deserialize handles.
+ * @param {string} requestName The human-friendly name of the request.
+ */
+function wrapBidi(handlerSet, requestName) {
+  shimmer.wrap(handlerSet, 'func', function (func) {
+    return function serverMethodTrace(stream) {
+      var that = this;
+      var args = arguments;
+      // Running in the namespace here propagates context to func.
+      return api.runInRootSpan({
+        name: requestName,
+        skipFrames: 3
+      }, function(transaction) {
+        if (!transaction) {
+          return func.apply(that, args);
+        }
+        if (api.config.enhancedDatabaseReporting) {
+          shimmer.wrap(stream, 'sendMetadata', sendMetadataWrapper(transaction));
+        }
+        var spanEnded = false;
+        var endSpan = function() {
+          if (!spanEnded) {
+            spanEnded = true;
+            transaction.endSpan();
+          }
+        };
+        // Propagate context in stream event handlers.
+        api.wrapEmitter(stream);
+        // stream is a Duplex. Emitting a 'finish' or 'error' event
+        // suggests that no more data will be sent, so we end the span in these
+        // event handlers.
+        // Similar to client streams, the trace span should measure the time
+        // until the server has finished sending data back to the client, not
+        // the time that all data has been received from the client.
+        stream.on('finish', function() {
+          // End the span unless there is an error.
+          if (stream.status.code === 0) {
+            endSpan();
+          }
+        });
+        stream.on('error', function (err) {
+          if (!spanEnded && api.config.enhancedDatabaseReporting) {
+            transaction.addLabel('error', err);
+          }
+          endSpan();
+        });
+        return func.apply(that, args);
+      });
+    };
+  });
+}
+
+/**
+ * Returns a function that wraps the gRPC server register function in order
+ * to create trace spans for gRPC service methods.
+ * @param {Function} register The function Server.prototype.register
+ * @returns {Function} registerTrace The new wrapper function.
+ */
+function serverRegisterWrap(register) {
+  return function registerTrace(name, handler, serialize, deserialize, method_type) {
+    // register(n, h, s, d, m) is called in addService once for each service method.
+    // Its role is to assign the serialize, deserialize, and user logic handlers
+    // for each exposed service method. Here, we wrap these functions depending on the
+    // method type.
+    var result = register.apply(this, arguments);
+    var handlerSet = this.handlers[name];
+    var requestName = 'grpc:' + name;
+    // Proceed to wrap methods that are invoked when a gRPC service call is made.
+    // In every case, the function 'func' is the user-implemented handling function.
+    if (method_type === 'unary') {
```

Comment:
This might be more concisely represented as a switch statement

```JS
switch (method_type) {
  case 'unary':
    wrapUnary(handlerSet, requestName);
    break;
  case 'server_stream':
    wrapServerStream(handlerSet, requestName);
    break;
  case 'client_stream':
    wrapClientStream(handlerSet, requestName);
    break;
  case 'bidi':
    wrapBidi(handlerSet, requestName);
    break;
  default:
    api.logger.warn('gRPC Server: Unrecognized method_type ' + method_type);
}

return result;
```